roles/openshift-applier/tasks/process-template-with-params-from-vars.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Change params_from_vars into facts
+  set_fact:
+    param: "-p {{ item.key|upper }}={{ item.value }}"
+  with_dict: "{{ params_from_vars }}"
+  register: params_list_result
```

Comment:
Please update the above task with something like:
```
  - set_fact:
      command_line_params: "{{ command_line_params | default('') + ' -p ' + item.key + '=' + item.value }}"
    with_dict: "{{ params_from_vars }}"
```

This will allow you to remove the next 2 tasks and hence save on execution time.

Reply:
BTW: starting with Ansible 2.5, it is now recommended to use `loop` instead of the `with_X` style loops. More info on the change for `with_dict` here: https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html#with-dict

Reply:
this does not work (see below), it duplicates the last var. I spent a significant amount of time trying to get this to work in a single step, and debugging this in order to get it to work correctly. I've cleaned it up so it's down to two tasks instead of there. Let's keep this the way it is and then we can go back and refactor if you can find a way to do it in a single task.

I would use `loop`, but I currently don't have 2.5, and I can imagine several people are still using 2.4.  
```
TASK [debug] ***************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "-p USERNAME=testUsername -p PASSWORD=testPassword -p password=testPassword"
}
```

Reply:
I have this working in one-step. I'd be happy to assist with code changes here.

Reply:
please feel free to forward that code along, and i can test it again